### DISCUSSION TITLE
Is the STM32F427VIT6 supported

### DISCUSSION THREAD

**Discussion by natyshemer** [stm32duino/Arduino_Core_STM32#1455] [Q&A] [answered]
Hi, 
I am not sure if the STM32F427VIT6 is support with the Arduino framework. 
The board type for this MCU is board: armstrap_eagle2048 but I wonder if I can use the STM32F427ZG. 
The pinout should be the same I think so it's just a matter of the flash size. 
Is there any problem with that?

![image](https://user-images.githubusercontent.com/57321229/127184461-433a128d-0129-4714-b800-146339e4538e.png)

**Comment by ABOSTM** [comment, marked as answer]
Hi @natyshemer ,
You may face issue with pinout which are different for both MCU mentioned. You can see the difference by comparing their respective PeripheralPins.c: 
https://github.com/stm32duino/Arduino_Core_STM32/blob/master/variants/STM32F4xx/F427V(G-I)T_F429V(E-G-I)T_F437V(G-I)T_F439V(G-I)T/PeripheralPins.c
https://github.com/stm32duino/Arduino_Core_STM32/blob/master/variants/STM32F4xx/F427Z(G-I)T_F429ZET_F429Z(G-I)(T-Y)_F437Z(G-I)T_F439Z(G-I)(T-Y)/PeripheralPins.c

Some pins/alternate functions are available on STM32F427ZG, but not present on STM32F427VIT6.
And as you said there is also Flash which is double size on STM32F427VIT6 compare to STM32F427ZG.
So that should work to if you take care of those issues. (and maybe pay attention to upload method)

Alternatively:
With generic variant system introduced in core version 2.0.0 all MCU are supported, and in particular your MCU correspond to :
https://github.com/stm32duino/Arduino_Core_STM32/tree/master/variants/STM32F4xx/F427V(G-I)T_F429V(E-G-I)T_F437V(G-I)T_F439V(G-I)T
And you can add your board by following this wiki:
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29

**Reply by natyshemer** [reply]
This sounds really promising. I'll definitely try it out.
Thanks
